internal/client.go

```diff
@@ -626,6 +630,515 @@ type (
+		// Use GetSearchAttributes API to get valid key and corresponding value type.
+		SearchAttributes *commonpb.SearchAttributes
+	}
+
+	// ScheduleOptions configure the parameters for creating a schedule.
+	ScheduleOptions struct {
+		// ScheduleID - The business identifier of the schedule.
+		// Optional: defaulted to a uuid.
+		ScheduleID string
+
+		// Schedule - Describes when Actions should be taken.
+		Spec ScheduleSpec
+
+		// Action - Which Action to take.
+		Action ScheduleAction
+
+		// Overlap - Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+		// running. This can be changed after a Schedule has taken some Actions, and some changes might produce
+		// unintuitive results. In general, the later policy overrides the earlier policy.
+		//
+		// Optional: defaulted to SCHEDULE_OVERLAP_POLICY_SKIP
+		Overlap enumspb.ScheduleOverlapPolicy
+
+		// CatchupWindow - The Temporal Server might be down or unavailable at the time when a Schedule should take an Action.
+		// Optional: defaulted to 1 minute
+		CatchupWindow time.Duration
+
+		// PauseOnFailure - When an Action times out or reaches the end of its Retry Policy.
+		// Optional: defaulted to false
+		PauseOnFailure bool
+
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
+
+		// Paused - Start in paused state.
+		// Optional: defaulted to false
+		Paused bool
+
+		// RemainingActions - limit the number of Actions to take.
+		//
+		// This number is decremented after each Action is taken, and Actions are not
+		// taken when the number is `0` (unless {@link ScheduleHandle.trigger} is called).
+		//
+		// Optional: defaulted to zero
+		RemainingActions int64
+
+		// TriggerImmediately - Trigger one Action immediately.
+		// Optional: defaulted to false
+		TriggerImmediately bool
+
+		// ScheduleBackfill - Runs though the specified time periods and takes Actions as if that time passed by right now, all at once. The
+		// overlap policy can be overridden for the scope of the ScheduleBackfill.
+		ScheduleBackfill []ScheduleBackfill
+
+		// Memo - Optional non-indexed info that will be shown in list schedules.
+		Memo map[string]interface{}
+
+		// SearchAttributes - Optional indexed info that can be used in query of List schedules APIs (only
+		// supported when Temporal server is using ElasticSearch). The key and value type must be registered on Temporal server side.
+		// Use GetSearchAttributes API to get valid key and corresponding value type.
+		SearchAttributes map[string]interface{}
+	}
+
+	// ScheduleWorkflowExecution contains details on a workflows execution stared by a schedule.
+	ScheduleWorkflowExecution struct {
```

Comment:
I took this type from the proposed TS API. We could replace this type with `WorkflowExecution` . The difference is this type highlights the runID is the original run ID.

Reply:
>we don't expose WorkflowExecution to users anyways, so this is fine.
We do in `WorkflowInfo` no?

Reply:
Yes, as `workflow.Execution`, but we should try to not require users use the `workflow` package outside of an actual workflow, and especially not for clients if we can help it. We messed up with `workflow.RegisterInfo` too. It is really hard for a user to discover from Go doc that some of this is visible from the `client` package and half from the `workflow` package.

Reply:
Oh I see your point, OK I'll leave as is.